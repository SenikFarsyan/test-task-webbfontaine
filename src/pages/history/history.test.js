import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import GameHistory from './history.component';

const mockStore = configureStore([]);

describe('connected react-redux GameHistory component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      user: {
        currentUser: {
          history: [
            { score: 50, date: '2020/04/21' },
            { score: 30, date: '2020/05/21' }
          ]
        }
      }
    });
    component = renderer.create(
      <Provider store={store}>
        <GameHistory />
      </Provider>
    );
  });

  it('should render with given state from Redux store', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
