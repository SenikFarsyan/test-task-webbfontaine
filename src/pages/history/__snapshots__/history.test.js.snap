// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`connected react-redux GameHistory component should render with given state from Redux store 1`] = `
<div
  className="container"
>
  <h3>
    Here are your scores
  </h3>
  <ol
    className="historyList"
  >
    <li>
      <span>
        50
         wpm
      </span>
       
      <span>
        in a month
      </span>
    </li>
    <li>
      <span>
        30
         wpm
      </span>
       
      <span>
        in 2 months
      </span>
    </li>
  </ol>
</div>
`;
